### ISSUE TITLE
Feature Request - Please support Raspberry Pi OS “Bookworm”

### ISSUE THREAD

**Issue by danmrossi** [reef-pi/reef-pi#2194]
Please support Raspberry Pi OS “Bookworm”

**Comment by 5ft24dave** [comment]
Since RPI foundation removed old methods of GPIO access, a temporary work around is:
`sudo  rpi-update d86b5843d68b9972a5430a6d3da1b271cfc83521`
this reverts back to Kernel 6.1.47, which still supports the older methods

**Comment by danmrossi** [comment]
Thanks for that, would raspberry pi 5 work on that kernel build do you know?

Sent from my iPhone

On 21 Apr 2024, at 4:05 AM, 5ft24dave ***@***.***> wrote:

﻿

Since RPI foundation removed old methods of GPIO access, a temporary work around is:
sudo rpi-update d86b5843d68b9972a5430a6d3da1b271cfc83521
this reverts back to Kernel 6.1.47, which still supports the older methods

**Comment by 5ft24dave** [comment]
Yes, it does. I have 2 Pi5's running the 6.1.74 kernel due to needing GPIO.

**Comment by danmrossi** [comment]
I get this error:

![image](https://github.com/reef-pi/reef-pi/assets/50608357/7b06c0ee-376d-48e8-b118-b2643e1265dd)

**Comment by 5ft24dave** [comment]
Pi updates just updated the kernel to 6.6.31, which kills GPIO again
If you just did an update, you need to run this again:
`sudo  rpi-update d86b5843d68b9972a5430a6d3da1b271cfc83521`

**Comment by danmrossi** [comment]
Yeah I did all the updates, rebooted and ran that prior to installing reef-pi
Sent from my iPhone

On 1 Jul 2024, at 10:40 PM, 5ft24dave ***@***.***> wrote:

﻿

Pi updates just updated the kernel to 6.6.31, which kills GPIO again
If you just did an update, you need to run this again:
sudo rpi-update d86b5843d68b9972a5430a6d3da1b271cfc83521

**Comment by 5ft24dave** [comment]
What does `uname -a` show?

**Comment by danmrossi** [comment]
Linux RoboCtrl 6.1.74-v8-16k+ #1725 SMP PREEMPT Mon Jan 22 13:38:39 GMT 2024 aarch64 GNU/Linux

RoboCtrl being the name of the device

**Comment by danmrossi** [comment]
my config.txt:

# For more options and information see
# http://rptl.io/configtxt
# Some settings may impact device functionality. See link above for details

# Uncomment some or all of these to enable the optional hardware interfaces
dtparam=i2c_arm=on
dtparam=i2c_arm_baudrate=10000
dtoverlay=pwm-2chan
#dtparam=i2s=on
dtparam=spi=off

# Enable audio (loads snd_bcm2835)
dtparam=audio=on

# Additional overlays and parameters are documented
# /boot/firmware/overlays/README

# Automatically load overlays for detected cameras
camera_auto_detect=1

# Automatically load overlays for detected DSI displays
display_auto_detect=1

# Automatically load initramfs files, if found
auto_initramfs=1

# Enable DRM VC4 V3D driver
dtoverlay=vc4-kms-v3d
max_framebuffers=2

# Don't have the firmware create an initial video= setting in cmdline.txt.
# Use the kernel's default instead.
disable_fw_kms_setup=1

# Run in 64-bit mode
arm_64bit=1

# Disable compensation for displays with overscan
disable_overscan=1

# Run as fast as firmware / board allows
arm_boost=1

[cm4]
# Enable host mode on the 2711 built-in XHCI USB controller.
# This line should be removed if the legacy DWC2 controller is required
# (e.g. for USB device mode) or if USB support is not required.
otg_mode=1

[all]
dtoverlay=w1-gpio,gpiopin=4
dtoverlay=w1-gpio,gpiopin=22
dtoverlay=i2c-rtc,ds1307
dtoverlay=disable-bt

**Comment by danmrossi** [comment]
`# For more options and information see
# http://rptl.io/configtxt
# Some settings may impact device functionality. See link above for details

# Uncomment some or all of these to enable the optional hardware interfaces
dtparam=i2c_arm=on
dtparam=i2c_arm_baudrate=10000
dtoverlay=pwm-2chan
#dtparam=i2s=on
dtparam=spi=off

# Enable audio (loads snd_bcm2835)
dtparam=audio=on

# Additional overlays and parameters are documented
# /boot/firmware/overlays/README

# Automatically load overlays for detected cameras
camera_auto_detect=1

# Automatically load overlays for detected DSI displays
display_auto_detect=1

# Automatically load initramfs files, if found
auto_initramfs=1

# Enable DRM VC4 V3D driver
dtoverlay=vc4-kms-v3d
max_framebuffers=2

# Don't have the firmware create an initial video= setting in cmdline.txt.
# Use the kernel's default instead.
disable_fw_kms_setup=1

# Run in 64-bit mode
arm_64bit=1

# Disable compensation for displays with overscan
disable_overscan=1

# Run as fast as firmware / board allows
arm_boost=1

[cm4]
# Enable host mode on the 2711 built-in XHCI USB controller.
# This line should be removed if the legacy DWC2 controller is required
# (e.g. for USB device mode) or if USB support is not required.
otg_mode=1

[all]
dtoverlay=w1-gpio,gpiopin=4
dtoverlay=w1-gpio,gpiopin=22
dtoverlay=i2c-rtc,ds1307
dtoverlay=disable-bt`

**Comment by 5ft24dave** [comment]
That *should* work.  Here is my uname -a output
`Linux AnneFish 6.1.74-v8-16k+ #1725 SMP PREEMPT Mon Jan 22 13:38:39 GMT 2024 aarch64 GNU/Linux`
this is working for 3 GPIO's for relays (lights, heater, Filter) and One-Wire for the temp sender.
My other tank for my Oscar is 2 heaters, 2 filters and lights, plus 2 one-wire temp senders.

**Comment by 5ft24dave** [comment]
I have another Pi5 sitting here I just got last month.  Let me go play with that one  the one I have working I got last October and started with the initial build of bookworm 64, and have just done the updates as they came out, and had to revert the kernel to get GPIO working.  Wonder if maybe there is something missing in the new distro that is also needed for GPIO besides the older kernel...

**Comment by 5ft24dave** [comment]
on the new one, installing bookworm fresh, and release 6 of reef-pi, I load my current database from my Oscar tank, all looks good, until I try to switch any piece of equipment, then I get:
`{"error":"Failed to update. Error: can't update outlet Outlet 4 driver lookup failure: driver by id rpi not available - can't get output pin"} | HTTP 500`

not sure why this isn't working here, but works on the other 2...

**Comment by 5ft24dave** [comment]
Well, Investigating more... The 2 that I have had operating, if I try to switch anything I get the same error as above... All relays I have switched to reverse so de-energized is on, energized is off.  When these were setup, I uploaded the database from the Pi4 that was running each one previously  DB import had no failures.  only error I have had is `failed to initialize driver: LED Driver Error: remote I/O error` which I don't use the LED driver for dimming etc, just relay to turn the lights off and on.  Unfortunately, with my tanks in the garage, I have never been in there early morning or late night, so haven't seen that the lighting didn't switch off...  I guess its a good thing my heaters are calibrated and have been keeping the temp where I need it...  back to the Pi4's until this can be fixed

**Comment by danmrossi** [comment]
Ahh so it’s not only me, thanks for trying I’ll dig through some code and see if I can work out anything - @ranjib any ideas come to mind?

Sent from my iPhone

On 2 Jul 2024, at 3:23 AM, 5ft24dave ***@***.***> wrote:

﻿

Well, Investigating more... The 2 that I have had operating, if I try to switch anything I get the same error as above... All relays I have switched to reverse so de-energized is on, energized is off. When these were setup, I uploaded the database from the Pi4 that was running each one previously DB import had no failures. only error I have had is failed to initialize driver: LED Driver Error: remote I/O error which I don't use the LED driver for dimming etc, just relay to turn the lights off and on. Unfortunately, with my tanks in the garage, I have never been in there early morning or late night, so haven't seen that the lighting didn't switch off... I guess its a good thing my heaters are calibrated and have been keeping the temp where I need it... back to the Pi4's until this can be fixed

**Comment by danmrossi** [comment]
Im thinking maybe we have to compile a version of 6.0 on Bookworm OS using the latest github commits? However we do that,
